android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUITabLayoutTabGroup.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2012 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+package ti.modules.titanium.ui.widget.tabgroup;
+
+import android.graphics.drawable.Drawable;
+import android.support.design.widget.TabLayout;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.TiBaseActivity;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.util.TiColorHelper;
+import org.appcelerator.titanium.view.TiCompositeLayout;
+import org.appcelerator.titanium.view.TiDrawableReference;
+
+import ti.modules.titanium.ui.TabGroupProxy;
+import ti.modules.titanium.ui.TabProxy;
+
+/**
+ * TabGroup implementation using TabLayout as a controller.
+ * This clas has been created for a backward compatibility with versions
+ * that relied on the implementation based on the deprecated ActionBar tab
+ * navigation mode.
+ *
+ * Functionality has been kept the same with minor visual differences which
+ * are introduced in favor of following Material Design guidelines.
+ */
+public class TiUITabLayoutTabGroup extends TiUIAbstractTabGroup implements TabLayout.OnTabSelectedListener
+{
+	// region private fields
+	private TabLayout mTabLayout;
+	// endregion
+
+	public TiUITabLayoutTabGroup(TabGroupProxy proxy, TiBaseActivity activity)
+	{
+		// Setup the action bar for navigation tabs.
+		super(proxy, activity);
+	}
+
+	/**
+	 * Removes the controller from the UI layout.
+	 * @param disable
+	 */
+	@Override
+	public void disableTabNavigation(boolean disable)
+	{
+		super.disableTabNavigation(disable);
+		if (disable) {
+			this.mTabLayout.setVisibility(View.GONE);
+		} else {
+			this.mTabLayout.setVisibility(View.VISIBLE);
+		}
+	}
+
+	@Override
+	public void addViews(TiBaseActivity activity)
+	{
+		this.mTabLayout = new TabLayout(activity);
+		// Set the colorPrimary as backgroundColor by default if do not have the backgroundColor set.
+		if (proxy.hasPropertyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
+			this.mTabLayout.setBackgroundColor(
+				TiColorHelper.parseColor(proxy.getProperty(TiC.PROPERTY_BACKGROUND_COLOR).toString()));
+		} else {
+			this.mTabLayout.setBackgroundColor(this.colorPrimaryInt);
+		}
```

Comment:
We need to use `PROPERTY_TABS_BACKGROUND_COLOR` here, not `PROPERTY_BACKGROUND_COLOR`.